calicoctl/commands/migrateipam.go

```diff
@@ -0,0 +1,301 @@
+		// If the resource does not exist, do not throw the error
+		if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+			return err
+		}
+	}
+
+	// Convert all of the abstract KV Pairs into the appropriate types.
+	blocks := []*IPAMBlockKVPair{}
+	for _, item := range blockKVList.KVPairs {
+		modelBlockKey, ok := item.Key.(model.BlockKey)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to a BlockKey", item.Key)
+		}
+
+		// Convert this to a key that has values for json encoding.
+		blockKey := BlockKey{
+			CIDR: modelBlockKey.CIDR,
+		}
+
+		block, ok := item.Value.(*model.AllocationBlock)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to an AllocationBlock", item.Value)
+		}
+		blocks = append(blocks, &IPAMBlockKVPair{
+			Key:      &blockKey,
+			Value:    block,
+			Revision: item.Revision,
+			TTL:      item.TTL,
+		})
+	}
+
+	blockAffinities := []*BlockAffinityKVPair{}
+	for _, item := range blockAffinityKVList.KVPairs {
+		modelBlockAffinityKey, ok := item.Key.(model.BlockAffinityKey)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to a BlockAffinityKey", item.Key)
+		}
+
+		// Convert this to a key that has vlaues for json encoding.
+		blockAffinityKey := BlockAffinityKey{
+			CIDR: modelBlockAffinityKey.CIDR,
+			Host: modelBlockAffinityKey.Host,
+		}
+
+		blockAffinity, ok := item.Value.(*model.BlockAffinity)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to a BlockAffinity", item.Value)
+		}
+		blockAffinities = append(blockAffinities, &BlockAffinityKVPair{
+			Key:      &blockAffinityKey,
+			Value:    blockAffinity,
+			Revision: item.Revision,
+			TTL:      item.TTL,
+		})
+	}
+
+	ipamHandles := []*IPAMHandleKVPair{}
+	for _, item := range ipamHandleKVList.KVPairs {
+		handleKey, ok := item.Key.(model.IPAMHandleKey)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to an IPAMHandleKey", item.Key)
+		}
+		handle, ok := item.Value.(*model.IPAMHandle)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to an IPAMHandle", item.Value)
+		}
+		ipamHandles = append(ipamHandles, &IPAMHandleKVPair{
+			Key:      &handleKey,
+			Value:    handle,
+			Revision: item.Revision,
+			TTL:      item.TTL,
+		})
+	}
+
+	var ipamConfig *IPAMConfigKVPair
+	if ipamConfigKV != nil {
+		configKey, ok := ipamConfigKV.Key.(model.IPAMConfigKey)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to an IPAMConfigKey", ipamConfigKV.Key)
+		}
+		config, ok := ipamConfigKV.Value.(*model.IPAMConfig)
+		if !ok {
+			return fmt.Errorf("Could not convert %+v to an IPAMConfig", ipamConfigKV.Value)
+		}
+		ipamConfig = &IPAMConfigKVPair{
+			Key:      &configKey,
+			Value:    config,
+			Revision: ipamConfigKV.Revision,
+			TTL:      ipamConfigKV.TTL,
+		}
+	}
+
+	// Store the information
+	m.BlockAffinities = blockAffinities
+	m.IPAMBlocks = blocks
+	m.IPAMHandles = ipamHandles
+	m.IPAMConfig = ipamConfig
+	return nil
+}
+
+// TODO: Cannot use the backend client for this. Need a better way
```

Comment:
why not? Is this comment right?

Reply:
comment isn't correct and I forgot to remove it.